File: assignment2.py
import time


def iterative_factorial(number):
    factorial = 1
    for i in range(1, number + 1):
        factorial *= i
    return factorial


def recursive_factorial(n):
    if n == 0:
        return 1
    else:
        return n * recursive_factorial(n - 1)


values = [5,10,50,70,100,300,500,800,1000,2000]


for num in values:
   
    start_iterative = time.time()
    result_iterative = iterative_factorial(num)
    end_iterative = time.time()
    duration_iterative = (end_iterative - start_iterative) * 1000  

    start_recursive = time.time()
    result_recursive = recursive_factorial(num)
    end_recursive = time.time()
    duration_recursive = (end_recursive - start_recursive) * 1000  

    print(f"Number: {num}")
    print(f"  Iterative Factorial: {result_iterative}")
    print(f"  Iterative Time: {duration_iterative:.3f} milliseconds")
    print(f"  Recursive Factorial: {result_recursive}")
    print(f"  Recursive Time: {duration_recursive:.3f} milliseconds")
    print("-----------------------------")

File: assignment2.js
function iterativeFactorial(number) {
    let factorial = 1;
    for (let i = 1; i <= number; i++) {
        factorial *= i;
    }
    return factorial;
}

function recursiveFactorial(n) {
    if (n === 0) {
        return 1;
    } else {
        return n * recursiveFactorial(n - 1);
    }
}

let values =
[5,10,50,70,100,300,500,800,1000,10000,15000];

for (let i = 0; i < values.length; i++) {
    let num = values[i];

    let startIterative = performance.now();
    let resultIterative = iterativeFactorial(num);
    let endIterative = performance.now();
    let durationIterative = endIterative - startIterative;

    let startRecursive = performance.now();
    let resultRecursive = recursiveFactorial(num);
    let endRecursive = performance.now();
    let durationRecursive = endRecursive - startRecursive;

    console.log("Number: " + num);
    console.log("  Iterative Factorial: " + resultIterative);
    console.log("  Iterative Time: " + durationIterative.toFixed(3) + " milliseconds");

    console.log("  Recursive Factorial: " + resultRecursive);
    console.log("  Recursive Time: " + durationRecursive.toFixed(3) + " milliseconds");

    console.log("-----------------------------");
}
